float32 angle
bool angle_provided